cloudmarker/alerts/emailalert.py

```diff
@@ -12,98 +12,86 @@
-            password (str): password of email account
+            from_addr (str): Sender's email address.
+            to_addrs (list): A list of :obj:`str` objects where each
+                :obj:`str` object is a recipient's email address.
+            host (str): SMTP host.
+            port (int): SMTP port.
+            use_ssl (bool): Use SSL if ``True``, not otherwise.
+            username (str): SMTP username.
+            password (str): SMTP password.
         """
         self._host = host
         self._port = port
         self._subject = subject
-        self._to = to
-        self._sender = sender
-        self._body = body
+        self._from_addr = from_addr
+        self._to_addrs = to_addrs
         self._stringbuffer = []
         self._use_ssl = use_ssl
         self._username = username
         self._password = password
 
     def write(self, record):
-        """Write JSON records to the file system.
-
-        This method is called once for every ``record`` read from a
-        cloud. In this example implementation of a alert, we simply
-        send the ``record`` in JSON format via email to the recipient.
-        The records keeps appending to a stringbuffer which then is
-        written in the email body
+        """Save event record in a buffer.
 
         Arguments:
-            record (dict): Data to send via email.
+            record (dict): An event record.
         """
         for _, value in record.items():
             self._stringbuffer.append(repr(value))
 
     def done(self):
-        """Perform final cleanup tasks.
-
-        This method is called after all records have been written. In
-        this example implementation, we properly terminate the JSON
-        array in the email body.
-
-        The connection is made based whether the SMTP or SMTP_SSL is
-        required which is determined by ``use_ssl`` param. The SMTP
-        login process is conducted if ``username`` param is provided.
-
-        """
+        """Send the buffered events as an email alert."""
         message = MIMEMultipart()
         message['Date'] = email.utils.formatdate(localtime=True)
         message['Subject'] = self._subject
-        message['From'] = self._sender
-        message['To'] = email.utils.COMMASPACE.join(self._to)
+        message['From'] = self._from_addr
+        message['To'] = email.utils.COMMASPACE.join(self._to_addrs)
 
-        # Incase if the buffer is empty then the default message will be sent
-        if self._stringbuffer:
-            self._body = ''.join(self._stringbuffer)
-        message.attach(MIMEText(self._body))
+        body = ''.join(self._stringbuffer)
+        message.attach(MIMEText(body))
 
         smtp = self._prepare_smtp_session()
         try:
-            smtp.sendmail(self._sender, self._to, message.as_string())
+            smtp.sendmail(self._from_addr,
+                          self._to_addrs,
+                          message.as_string())
```

Comment:
Most of the other plugins have a success message logged which I personally find very convenient to verify if the method actually succeeded. The other alert, however, does not have it as well. You may choose to add it if you feel it is required.

Reply:
I agree. I was trying to keep this change minimal, i.e., just focus on the documentation and necessary code updates. This plugin requires quite a bit of rework to move the SMTP code out to `util` module and make it consistent with the other plugins. I will do some of it as part of #53. I will address this suggestion of yours while resolving #53.